### DISCUSSION TITLE
How to validate type if tag contains valid type of data?

### DISCUSSION THREAD

**Discussion by qarmin** [Enet4/dicom-rs#559] [Q&A] [answered]
Recently I found that in one place I used SH instead CS type.

I wanted to create simple validator to check if object is proper type.

I found ENTRIES slice, that could be used, but I cannot use it outside of this crate

https://github.com/Enet4/dicom-rs/blob/4f2c0d8d0857e808f59b5cc7840b4a0dde550ab5/dictionary-std/src/tags.rs#L16005C18-L16005C25

Is there any other way to validate object type?

**Comment by Enet4** [comment, marked as answer]
You can query the data element dictionary through [`dicom_dictionary_std::StandardDataDictionary`](https://docs.rs/dicom-dictionary-std/latest/dicom_dictionary_std/data_element/struct.StandardDataDictionary.html), [`DataDictionary`](https://docs.rs/dicom-core/latest/dicom_core/dictionary/trait.DataDictionary.html) trait. Admittedly, we're a bit thin on examples here.

```rust
use dicom_core::prelude::*;
use dicom_dictionary_std::{tags, StandardDataDictionary};

let entry = StandardDataDictionary.by_tag(tags::MODALITY);
let vr = entry.vr();
assert_eq!(vr.exact(), Some(VR::CS));
```

(the `vr.exact()` is needed here because dictionary entries have a virtual VR type for scenarios where the VR depends on context)
